### PR TITLE
452 principalinvestigator

### PR DESCRIPTION
**Pull request by eriktks**
PI has been removed from related projects block, both in software listing and project listing, cf issue #575 . Additionally, code has been added so that information about the teams of related projects can be displayed on the website in the future cf. issue #452 . In that case, the following code snippet can be added to the templates (example for principal investigator):
```
<div class="author">
    <div class="author_name">
        {% for team_member in project.team %}
            {% if team_member.role == 'Principal investigator' %}
                {{ team_member.givenNames }}
                {{ team_member.nameParticle }}
                {{ team_member.familyNames }}
                {{ team_member.nameSuffix }}
            {% endif %}
        {% endfor %}
    </div>
</div>
```

### CODE REVIEW

File: frontend/app/application.py [hunk 1]
```diff
@@ -103,12 +103,26 @@ def set_markdown(software, fields):
...
 
 
+def fetch_team_member_data(team_member):
+    person_id = team_member['foreignKey']['id']
+    url = api_url + "/person/%s" % person_id
+    person_dictionary = requests.get(url).json()
+    if "error" in person_dictionary:
+        return team_member
+    team_member.update(person_dictionary)
+    return team_member
```
Comment (jspaaks) [inline frontend/app/application.py:113]: Not entirely sure I understand your intent correctly, but I think this work is already done in https://github.com/research-software-directory/research-software-directory/blob/9d1ef5625366d1a322ad052ea6dd3794745eee90/harvesting/cache.py#L61-L66

File: frontend/app/application.py [hunk 2]
```diff
@@ -103,12 +103,26 @@ def set_markdown(software, fields):
...
+ 
+
 @application.route('/software/<software_id>')
 def software_product_page_template(software_id):
     url = api_url + "/software_cache/%s" % software_id
     software_dictionary = requests.get(url).json()
     if "error" in software_dictionary:
         return flask.redirect("/", code=302)
+    for i in range(0,len(software_dictionary['related']['projects'])):
```
Comment (jspaaks) [inline frontend/app/application.py:122]: Just a quick observation: 

Is there a reason why you're introducing a loop counter? If not, consider using something like

```suggestion
    for project in software_dictionary['related']['projects']:
```

and similarly,

```
for member in project['foreignKey']['team']
```

File: frontend/app/application.py (hunk 1 above)
Comment (eriktks) [reply frontend/app/application.py:113]: The goal of the function fetch_team_member_data is to get the name strings of the people in the team of the project. The function cache_project retrieves the data of a project using its key but does not resolve the foreign keys in the team data within the project data structure.

The name strings were necessary for solving issue #452  but they were made unnecessary by issue #575 . So this additional code can also be removed if it causes confusion.

File: frontend/app/application.py (hunk 2 above)
Comment (eriktks) [reply frontend/app/application.py:122]: The reason was the assignment inside the loop. I did not know that you could change a dictionary by changing the value of a a variable that was an alias for a part inside the dictionary. The code was updated and committed (e70c106). It is much more readable now. Thanks for the tip!

**Comment by sverhoeven** [comment]
Can you split the PR into 2 PRs
1. For #575
1. For #452 

That way we dont have unused code.

**Comment by eriktks** [comment]
Good suggestion, @sverhoeven . This PR will be closed. The followup PR is #589 (for issue #575 ). Issue #452 has been made obsolete by #575 and does not require any action.

**Pull request closed without merging**